### PR TITLE
add warnings for deprecated packages

### PR DESCRIPTION
**Pull request by dirk-thomas**
Add warnings when building a deprecated package in a workspace and when find_package()-ing it (using a deprecated package).

### CODE REVIEW

File: cmake/parse_package_xml.py [hunk 1]
```diff
@@ -25,6 +25,9 @@ def _get_output(package):
     values['BUILD_DEPENDS'] = ' '.join(['"%s"' % str(d) for d in package.build_depends])
     values['RUN_DEPENDS'] = ' '.join(['"%s"' % str(d) for d in package.run_depends])
 
+    deprecated = [e.content for e in package.exports if e.tagname == 'deprecated']
+    values['DEPRECATED'] = '"%s"' % ((deprecated[0] if deprecated else 'TRUE') if deprecated else '')
```
Comment (wjwwood) [inline cmake/parse_package_xml.py]: Isn't the second `if deprecated else ''` redundant?

Comment (wjwwood) [inline cmake/parse_package_xml.py]: Should that be:

```
values['DEPRECATED'] = '"%s"' % ((deprecated[0] if deprecated[0] else 'TRUE') if deprecated else '')
```

Comment (dirk-thomas) [inline cmake/parse_package_xml.py]: exactly - updated.

**Comment by wjwwood** [comment]
Looks good to me.

**Pull request merged**